src/adapter/breakpoints.ts

```diff
@@ -94,6 +94,15 @@ export class BreakpointManager {
     urlUtils.lowerCaseInsensitivePath,
   );
 
+  public hasAtLocation(location: IUiLocation) {
+    const breakpointsAtPath = this._byPath.get(location.source.absolutePath()) || [];
```

Comment:
I would suggest checking _byRef on the source as well. That code path isn't used in these changes, but that would avoid a footgun if someone tries to use this later and wonders why it isn't working. Something like

```ts
return [this._byRef.get(location.source.sourceReference(), this._byPath.get(location.source.absolutePath)]
    .some(breakpoints => breakpoints?.some(bp => ...
```